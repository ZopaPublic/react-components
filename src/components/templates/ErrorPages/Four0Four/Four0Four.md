### Summary

`<ErrorPageTemplate.Four0Four />` renders the default content for when a page cannot be found

### Example

```tsx
import { ErrorPageTemplate } from '@zopauk/react-components';

<ErrorPageTemplate>
  <ErrorPageTemplate.Four0Four />
</ErrorPageTemplate>;
```
