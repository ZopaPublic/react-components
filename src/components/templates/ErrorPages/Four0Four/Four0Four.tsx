import React from 'react';
import { faInfo } from '@fortawesome/free-solid-svg-icons';

import FlexRow from '../../../layout/FlexRow/FlexRow';
import FlexCol from '../../../layout/FlexCol/FlexCol';
import Heading from '../../../atoms/Heading/Heading';
import Text from '../../../atoms/Text/Text';
import { useViewport } from '../../../../hooks/useViewport/useViewport';
import { breakpoints } from '../../../../constants/breakpoints';
import { colors } from '../../../../constants';
import { ErrorTemplateProps } from '../types';
import { StyledIcon, StyledLink } from '../styles';
import { ThemeProvider, useThemeContext } from '../../../styles/Theme';

export interface Four0FourErrorProps extends ErrorTemplateProps {
  linkUrl?: string;
  linkText?: string;
}

const Four0Four = ({
  icon = faInfo,
  linkUrl = 'https://www.zopa.com',
  linkText = 'Go to Zopa home',
}: Four0FourErrorProps) => {
  const { width = 0 } = useViewport();
  const theme = useThemeContext();

  return (
    <div data-automation="ZA.ErrorPage.Four0Four">
      <ThemeProvider theme={theme}>
        <FlexRow justify="center">
          <FlexCol xs="auto" className="mb-7">
            <StyledIcon bgColor={colors.greyLighter} variant={icon} size="3x" color={colors.greyDark} />
          </FlexCol>
        </FlexRow>
        <Heading as="h1" size={width <= breakpoints.desktop ? 'h4' : 'h3'} align="center" className="mb-4">
          Sorry, there’s been a problem
        </Heading>
        <Text as="p" size={width <= breakpoints.desktop ? 'body' : 'lead'} align="center" className="mb-8">
          Seems like the page you’re looking for doesn’t exist.
        </Text>
        <FlexRow justify="center">
          <FlexCol xs="auto">
            <StyledLink className="mb-10" href={linkUrl} styling="secondary">
              {linkText}
            </StyledLink>
          </FlexCol>
        </FlexRow>
        <Text as="p" size="small" align="center">
          Error code: 404
        </Text>
      </ThemeProvider>
    </div>
  );
};

export default Four0Four;
