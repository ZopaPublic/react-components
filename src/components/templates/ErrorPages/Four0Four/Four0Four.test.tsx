import React from 'react';
import { render } from '@testing-library/react';

import Four0Four from './Four0Four';

describe('<Four0Four />', () => {
  it('renders with all the props', () => {
    const { container } = render(<Four0Four />);
    expect(container).toMatchSnapshot();
  });
});
