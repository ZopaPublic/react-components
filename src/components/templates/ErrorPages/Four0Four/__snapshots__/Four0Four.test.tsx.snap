// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Four0Four /> renders with all the props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: -12px;
  margin-left: -12px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: 12px;
  padding-left: 12px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c2 {
  background-color: #EFEFEF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 24px;
}

.c4 {
  font-size: 24px;
  line-height: 32px;
  -webkit-letter-spacing: -0.25px;
  -moz-letter-spacing: -0.25px;
  -ms-letter-spacing: -0.25px;
  letter-spacing: -0.25px;
  margin: 0;
  color: #2C3236;
  text-align: center;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 800;
}

.c5 {
  margin: 0;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #2C3236;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  text-align: center;
}

.c8 {
  margin: 0;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #2C3236;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  text-align: center;
}

.c6 {
  background-color: transparent;
  font-size: inherit;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  line-height: inherit;
  color: #3B46C4;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  border: none;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  color: auto;
}

.c6:hover,
.c6:active {
  color: #21247F;
}

.c6:hover svg path,
.c6:active svg path {
  fill: #21247F;
}

.c6:hover {
  color: auto;
}

.c6:hover svg path {
  fill: auto;
}

.c6:active {
  color: auto;
}

.c6:active svg path {
  fill: auto;
}

.c3 {
  width: 80px;
  height: 80px;
}

.c7 {
  outline: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 12px 24px;
  background: #E9EAFA;
  border-radius: 8px;
  border: 1px solid transparent;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  color: #21247F;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.c7:focus:not(:active) {
  border: 1px solid #FFFFFF;
  box-shadow: 0 0 4px #3B46C4;
}

.c7:active {
  opacity: 0.8;
  box-shadow: unset;
  border: 1px solid transparent;
}

.c7:hover {
  background: #EEEFFB;
}

@media (min-width:0px) {
  .c1 {
    display: block;
    -webkit-flex: 0 0 auto;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    width: auto;
    max-width: none;
  }
}

<div>
  <div
    data-automation="ZA.ErrorPage.Four0Four"
  >
    <div
      class="c0"
      cols="12"
      direction="row"
    >
      <div
        class="c1 mb-7"
        cols="12"
      >
        <div
          class="c2 c3"
          color="#EFEFEF"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-info fa-w-6 fa-3x "
            color="#4A545E"
            data-icon="info"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 192 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <h1
      class="c4 mb-4"
    >
      Sorry, there’s been a problem
    </h1>
    <p
      class="c5 mb-8"
    >
      Seems like the page you’re looking for doesn’t exist.
    </p>
    <div
      class="c0"
      cols="12"
      direction="row"
    >
      <div
        class="c1"
        cols="12"
      >
        <a
          class="c6 c7 mb-10"
          color="#3B46C4"
          href="https://www.zopa.com"
        >
          Go to Zopa home
        </a>
      </div>
    </div>
    <p
      class="c8"
    >
      Error code: 404
    </p>
  </div>
</div>
`;
